app/src/main/java/org/odk/odknotifications/Activities/MainActivity.java

```diff
@@ -364,10 +363,11 @@ public void onSuccess(PendingDynamicLinkData pendingDynamicLinkData) {
                                     public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
                                         Group group = new Group((String)dataSnapshot.child("id").getValue(),(String)dataSnapshot.child("name").getValue(),0);
                                         new SubscribeNotificationGroup(MainActivity.this,group.getId(),getActiveUser()).execute();
-                                        DatabaseReference databaseReference = FirebaseDatabase.getInstance().getReference().child("clients").child(getActiveUser()).child("groups");
+                                        DatabaseReference databaseReference = FirebaseDatabase.getInstance().getReference().child("clients").child(getActiveUser()).child("groups").push();
                                         databaseReference.child("id").setValue(group.getId());
                                         databaseReference.child("name").setValue(group.getName());
```

Comment:
The changes look good but why is the Firebase database being modified here?

Reply:
Actually before, on subscribing to a new group, it was overwriting the previous group data on firebase. So, on syncing with the server it fetches only one group from the firebase database. To resolve this, each group data is now stored under a random key generated by `push()`